src/puppet-web/firer.ts

```diff
@@ -367,7 +366,7 @@ async function checkRoomTopic(m: Message): Promise<void> {
   if (/^You$/.test(changer) || /^你$/.test(changer)) {
     changerContact = Contact.load(this.userId)
   } else {
-    changerContact = room.member(changer)
+    changerContact = room.member({remark: changer}) || room.member({nick: changer})
```

Comment:
Why you copy/paste the same answer four times here?

I comment more than one times because I want to point out the position for you where needs to change.

What you were arguing about is not wrong, but it's not entirely right. Remember always to KISS(Keep It Simple, Stupid). 

The most scenario is we just want to get a member by name(whatever remark/nick/display). I do not agree with you that "it should work only with the function find the member by nickname", instead it should return a contact for any name.  And If "one set a remark same as nick", then there's the time to use the new QueryFilter as the parameter.

So just change it to what I said.